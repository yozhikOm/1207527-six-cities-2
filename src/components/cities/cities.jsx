import React from "react";
import PropTypes from 'prop-types';

const Cities = (props) => {
  const {cities, onCityClick} = props;
  return (
    <ul className="locations__list tabs__list">
      {cities.map((it, i) => (
        <li className="locations__item" key={`${i}`}>
          <a className="locations__item-link tabs__item" href="#" onClick={(evt) => {
            evt.preventDefault();
            let cityName = evt.currentTarget.textContent;
            let city = cities.find((c) => c.title === cityName);
            onCityClick(city);
          }}>
            <span>{it.title}</span>
          </a>
        </li>
      ))}
    </ul>
  );
};

Cities.propTypes = {
  cities: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired,
        coordinates: PropTypes.arrayOf(PropTypes.number, PropTypes.number).isRequired,
      })
  ).isRequired,
  onCityClick: PropTypes.func.isRequired,
};

export {Cities};
