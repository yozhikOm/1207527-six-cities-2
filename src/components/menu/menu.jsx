import React from 'react';
import PropTypes from 'prop-types';
import {Cities} from '../cities/cities.jsx';

const Menu = (props) => {
  const {cities, onCityClick} = props;
  return (
    <React.Fragment>
      <h1 className="visually-hidden">Cities</h1>
      <div className="tabs">
        <section className="locations container">
          <Cities cities={cities} onCityClick={onCityClick}/>
        </section>
      </div>
    </React.Fragment>
  );
};

Menu.propTypes = {
  cities: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired,
        coordinates: PropTypes.arrayOf(PropTypes.number, PropTypes.number).isRequired,
      })).isRequired,
  onCityClick: PropTypes.func.isRequired,
};

export {Menu};
