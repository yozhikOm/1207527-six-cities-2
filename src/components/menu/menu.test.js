import React from 'react';
import {Menu} from './menu.jsx';
import ShallowRenderer from 'react-test-renderer/shallow';
import {cities} from '../../mocks/cities.js';

it(`Menu компонент рендерится корректно`, () => {
  const renderer = new ShallowRenderer();

  renderer.render(<Menu
    cities={cities}
    onCityClick={jest.fn()}
  />);

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
