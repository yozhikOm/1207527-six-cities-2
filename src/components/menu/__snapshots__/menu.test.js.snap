// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu компонент рендерится корректно 1`] = `
<React.Fragment>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <Cities
        cities={
          Array [
            Object {
              "coordinates": Array [
                52.38333,
                4.9,
              ],
              "title": "Amsterdam",
            },
            Object {
              "coordinates": Array [
                52.52,
                13.405,
              ],
              "title": "Berlin",
            },
            Object {
              "coordinates": Array [
                40.7128,
                -74.006,
              ],
              "title": "New York",
            },
          ]
        }
        onCityClick={[MockFunction]}
      />
    </section>
  </div>
</React.Fragment>
`;
